### ISSUE TITLE
Compiling without sse/sse2 on x86_64 fails

### ISSUE THREAD

**Issue by Noodlez1232** [ziglang/zig#18159]
### Zig Version

0.12.0 (For some reason it didn't compile with the git version. Git hash: 7fbbeae)

### Reproduce
```sh
echo "pub fn main() void { }" > test.zig
zig build-exe -mcpu=x86_64-sse-sse2 test.zig
```
### Output
```
LLVM Emit Object... error: extendhfsf2.zig:16:5: in function __extendhfsf2 float (half): SSE register return with SSE disabled
````

### Expected Behavior

Compilation succeeds.

It's worth noting that this is not a regression and a similar error occurs on 0.11.0.

**Comment by rootbeer** [comment]
This can be reproduced slightly more directly with:

```
$ zig build-obj -mcpu=x86_64-sse-sse2 lib/compiler_rt/extendhfsf2.zig
LLVM Emit Object... error: extendhfsf2.zig:16:5: in function __extendhfsf2 float (half): SSE register return with SSE disabled
```

This is the code its doesn't like:
```
pub fn __extendhfsf2(a: common.F16T(f32)) callconv(.C) f32 {
    return extendf(f32, f16, @as(u16, @bitCast(a)));                        // <<<<<<< line 16
}
```

**Comment by Vexu** [comment]
Fixed by adding `+soft_float`, duplicate of #12227
